test/govspeak_images_test.rb

```diff
@@ -0,0 +1,61 @@
+# encoding: UTF-8
+
+require 'test_helper'
+require 'govspeak_test_helper'
+
+class GovspeakImagesTest < Minitest::Test
+  include GovspeakTestHelper
+
+  def build_image(attrs = {})
+    attrs[:alt_text] ||= "my alt"
+    attrs[:url] ||= "http://example.com/image.jpg"
+    attrs[:id] ||= "image-id"
+    attrs
+  end
+
+  test "Image:image-id syntax renders an image in options[:images]" do
+    given_govspeak "[Image:image-id]", images: [build_image] do
+      assert_html_output(
+        %{<figure class="image embedded">} +
+        %{<div class="img"><img src="http://example.com/image.jpg" alt="my alt"></div>} +
+        %{</figure>}
+      )
+    end
+  end
+
+  test "Image:image-id syntax escapes alt text" do
+    given_govspeak "[Image:image-id]", images: [build_image(alt_text: %{my alt '&"<>})] do
+      assert_html_output(
+        %{<figure class="image embedded">} +
+        %{<div class="img"><img src="http://example.com/image.jpg" alt="my alt '&amp;&quot;&lt;&gt;"></div>} +
+        %{</figure>}
+      )
+    end
+  end
+
+  test "Image:image-id syntax renders nothing if not found" do
+    doc = Govspeak::Document.new("[Image:another-id]")
+    assert_equal %{\n}, doc.to_html
+  end
+
+  test "Image:image-id syntax adds image caption if given" do
+    given_govspeak "[Image:image-id]", images: [build_image(caption: 'My Caption & so on')] do
```

Comment:
Do we know whether the whitehall caption equivalent supports or escapes HTML and that this is equivalent (hopefully it escapes)

Reply:
I don't think it does - it just calls directly with edition.images, and there doesn't seem to be any escaping on input. But surely ERB will do that automatically?

Reply:
Rails does it rather than ERB if I remember correctly. as this uses active support I'm not sure if ERB would be patched or not but I expect this would be vulnerable to outputting HTML (you can see the encode method on alt text and filename)

Reply:
Oops, forgot about the fun-filled textual render method ;-). No ERB to help there anyway - I'll add an encode to that as part of this.

Reply:
cool probably worth doing a cursory check that Whitehall doesn't pass govspeak through or anything - certainly hope not.

Reply:
Oh, that's an idea - but I tried creating a document with an image like that and the govspeak caption isn't processed. Makes some sense, actually, as otherwise you could have recursive govspeak extensions generating their own govspeak.